File: Giorno 5/esercizio1.js
function calcolaFattoriale(n) {
  if (n < 0) {
    return "Errore: il fattoriale non è definito per i numeri negativi.";
  } else if (n === 0) {
    return 1;
  } else {
    let prodotto = 1;
    for (let i = 1; i <= n; i++) {
      prodotto *= i;
    }
    return prodotto;
  }
}

const input = prompt("Inserisci un numero per calcolare il fattoriale:");
const N = parseInt(input);

if (isNaN(N)) {
  console.log("Input non valido. Inserisci un numero.");
} else {
  const risultato = calcolaFattoriale(N);
  console.log(`Il fattoriale di ${N} è: ${risultato}`);
}

File: Giorno 5/esercizio1.py
def calcola_fattoriale(n):
    if n < 0:
        return "Errore: il fattoriale non è definito per i numeri negativi."
    elif n == 0:
        return 1
    else:
        prodotto = 1
        for i in range(1, n + 1):
            prodotto *= i
        return prodotto

N = int(input("Inserisci un numero per calcolare il fattoriale: "))
risultato = calcola_fattoriale(N)

print(f"Il fattoriale di {N} è: {risultato}")